crates/uv-auth/src/middleware.rs

```diff
@@ -100,88 +140,199 @@ impl Middleware for AuthMiddleware {
         };
         trace!("Handling request for {url}");
 
-        // Then check for credentials in (2) the cache
-        let credentials = self.cache().check(request.url(), credentials);
-
-        // Track credentials that we might want to insert into the cache
-        let mut new_credentials = None;
-
-        // If already authenticated (including a password), don't query other services
-        if credentials.is_authenticated() {
-            match credentials {
-                // If we get credentials from the cache, update the request
-                CheckResponse::Cached(credentials) => request = credentials.authenticate(request),
-                // If we get credentials from the request, we should update the cache
-                // but don't need to update the request
-                CheckResponse::Uncached(credentials) => new_credentials = Some(credentials.clone()),
-                CheckResponse::None => unreachable!("No credentials cannot be authenticated"),
+        if let Some(credentials) = credentials {
+            let credentials = Arc::new(credentials);
+
+            // If there's a password, send the request and cache
+            if credentials.password().is_some() {
+                trace!("Request for {url} is already fully authenticated");
+                return self
+                    .complete_request(Some(credentials), request, extensions, next)
```

Comment:
I considered not caching in this case to reduce the size of the cache, but it's kind of confusing behavior and we use a trie to ensure cache performance is reasonable with many URLs.